### PR TITLE
dosc/sc5: rework installation docs

### PR DESCRIPTION
**Pull request by Choraden**
### Description

This patch makes install section for each of the platforms. 

Linux, macOS, and Windows instructions were rewritten. Docker and Kubernetes instructions were only moved, the content needs to be reworked.

Fixes SC-4583

### Motivation and Context

This gives a nice structure for installation docs. They are all in one place. Each one references one platform. 

### Types of Changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation fix (typos, incorrect content, missing content, etc.)

### CODE REVIEW

**Comment by Choraden** [comment]
> have to have a high-level doc available at https://docs.saucelabs.com/secure-connections/sauce-connect-5/installation/

Agree. Do you know if this Screenshot 2024-02-16 at 17 53 36 Installation section can have its content?

File: docs/secure-connections/sauce-connect-5/installation.md [hunk 1]
```diff
@@ -1,217 +1,12 @@
...
-See [this document](/secure-connections/sauce-connect/installation/) for the most recent Sauce Connect 4.x.x version download instructions.
-:::
-
-This topic describes how to install Sauce Connect Proxy version 5 to your machine.
-
-## What You’ll Need
-
-- A Sauce Labs account ([Log in](https://accounts.saucelabs.com/am/XUI/#login/) or sign up for a [free trial license](https://saucelabs.com/sign-up)).
-
-## Downloading Sauce Connect Proxy
```
Comment (waggledans) [inline docs/secure-connections/sauce-connect-5/installation.md:22]: there should be links to download binaries downloads, downloading a binary is still a legitimate way to install sc. I think that each OS, should have a subset of this table

File: docs/secure-connections/sauce-connect-5/installation.md [hunk 2]
```diff
@@ -1,217 +1,12 @@
...
-See [this guide](/secure-connections/sauce-connect-5/operation/kubernetes/)
-
-## More Information
-
-- [Sauce Connect Proxy Changelog](https://changelog.saucelabs.com/en?category=sauce%20connect)
-- [Sauce Connect Proxy 5 Administration](/secure-connections/sauce-connect-5/operation/overview/)
-- [Sauce Connect Proxy 5 CLI Reference](/dev/cli/sauce-connect-5/)
-- [Sauce Connect Proxy Architecture](/secure-connections/sauce-connect-5/advanced/architecture/)
+Visit the following pages for installation instructions for your platform:
+- [Docker](/secure-connections/sauce-connect-5/installation/docker)
```
Comment (waggledans) [inline docs/secure-connections/sauce-connect-5/installation.md:8]: maybe the order should be OSes first and then other:
- linux
- mac
- win
- docker
- k8s
We could add plugins there too, later: github action, Jenkins, etc...

File: docs/secure-connections/sauce-connect-5/installation/docker.md [hunk 3]
```diff
@@ -20,11 +20,11 @@ Here are some benefits/use cases for using containerized Sauce Connect Proxy:
 ## Running the Sauce Connect Docker Container
 
 1. Pull the Sauce Connect Proxy Docker Image from the [Docker Hub](https://hub.docker.com/r/saucelabs/sauce-connect).
-   - To pull the latest version of Sauce Connect Proxy (recommended):
+    - To pull the latest version of Sauce Connect Proxy (recommended):
```
Comment (waggledans) [inline docs/secure-connections/sauce-connect-5/installation/docker.md:23]: just noticed:
```
To pull the latest patch version of Sauce Connect Proxy 5.0
```

File: docs/secure-connections/sauce-connect-5/installation.md (hunk 1 above)
Comment (Choraden) [reply docs/secure-connections/sauce-connect-5/installation.md:22]: All the links are still there. They can be found in specific instructions, e.g.
Screenshot 2024-02-19 at 08 32 04

File: docs/secure-connections/sauce-connect-5/installation.md (hunk 2 above)
Comment (Choraden) [reply docs/secure-connections/sauce-connect-5/installation.md]: ok

File: docs/secure-connections/sauce-connect-5/installation/docker.md (hunk 3 above)
Comment (Choraden) [reply docs/secure-connections/sauce-connect-5/installation/docker.md]: The docker instructions will be completely reworked.

File: docs/secure-connections/sauce-connect-5/installation.md (hunk 1 above)
Comment (waggledans) [reply docs/secure-connections/sauce-connect-5/installation.md:22]: I think the table containing links is useful - you can click on the link in any browser and the binary gets downloaded.

**Review by waggledans — APPROVED** [review]

**Pull request merged**